opencti-platform/opencti-front/src/utils/filters/useFiltersState.tsx

```diff
@@ -0,0 +1,118 @@
+import { useState } from 'react';
+import { v4 as uuid } from 'uuid';
+import { emptyFilterGroup, Filter, FilterGroup } from './filtersUtils';
+import {
+  handleAddFilterWithEmptyValueUtil,
+  handleAddRepresentationFilterUtil,
+  handleAddSingleValueFilterUtil,
+  handleChangeOperatorFiltersUtil,
+  handleClearAllFiltersUtil,
+  handleRemoveFilterUtil,
+  handleRemoveRepresentationFilterUtil,
+  handleSwitchLocalModeUtil,
+} from './filtersManageStateUtil';
+import { handleFilterHelpers } from '../hooks/useLocalStorage';
+
+interface useFiltersStateProps {
+  filters: FilterGroup,
+  latestAddFilterId?: string
+}
+const useFiltersState = (initFilters: FilterGroup = emptyFilterGroup): [FilterGroup, handleFilterHelpers] => {
+  const [filtersState, setFiltersState] = useState<useFiltersStateProps>({
+    filters: initFilters,
+    latestAddFilterId: undefined,
+  });
+  const helpers: handleFilterHelpers = {
```

Comment:
Those helpers are re-renders every time the hook is rendered. I don't know if it's really an issue here but IMO it could be a case where `memo` has benefits.

An other point disturbing me is the fact to use `filterState` to call `setFiltersState` in helpers functions. Again maybe there is no issue right now but could lead to hard debugging ones later maybe. Have a look at https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

Reply:
Good point. I will look into it and let you know. Thanks 👍

Reply:
Instead of calling filterState, I am calling prevState now. Very nice call.
For the first point, I am not sure how we should optimize with `memo` in this case.
Maybe `useCallback `on each helpers function ?